app/components/user-cards.js

```diff
@@ -59,23 +54,75 @@ export default Component.extend({
 
         this.set('sort_option','first_name')
         this.set('sort_order','asc')
+        this.set('sort_title','First Name')
+    },
 
+    searchByEmployeeName(){
+        if (this.search !== '') {
+            this.set('displaysearch',this.search);
+            }
+        else {
+            this.set('displaysearch',null);
+        }
+    },   
+
+    filterArrayElements(arrayItems,search){
+
+            if((get(arrayItems,'fullName')).toLowerCase().startsWith(search.toLowerCase())  || 
+            (get(arrayItems,'last_name')).toLowerCase().startsWith(search.toLowerCase()) ){
+               return arrayItems;
+         }
+            else {
+                return false;
+         }
     },
 
+    sortEmployeeList(arrayName,order,option,searchemployee){
+        
+        if(order=='asc'){
+            if(searchemployee!=null){
+               return arrayName.filter((element)=> this.filterArrayElements(element,searchemployee)).sortBy(option)
+            }
+            else
+            return arrayName.sortBy(option)
+
+        }
+        else {
+            if(searchemployee!=null){
+                return arrayName.filter((element)=> this.filterArrayElements(element,searchemployee)).sort(
+                    (item1, item2) =>(get(item1,option) < get(item2,option)) ? 1 : (get(item1,option) > 
+                        get(item2,option)) ? -1 : 0);
+            }
+            else{
+            return arrayName.sort(
+                (item1, item2) =>(get(item1,option) < get(item2,option)) ? 1 : (get(item1,option) > 
+                    get(item2,option)) ? -1 : 0);
+            }
+
```

Comment:
Instead of we check for the condition `searchemployee!=null` in both if and else, we can move it inside `filterArrayElements` method.